usr/lib/common/loadsave.c

```diff
@@ -2415,20 +2474,14 @@ CK_RV load_private_token_objects(STDLL_TokData_t *tokdata)
     if (tokdata->version < TOK_NEW_DATA_STORE)
         return load_private_token_objects_old(tokdata);
 
-    sprintf(iname, "%s/%s/%s", tokdata->data_store, PK_LITE_OBJ_DIR,
-            PK_LITE_OBJ_IDX);
-
-    fp1 = fopen(iname, "r");
+    fp1 = open_token_object_index(iname, sizeof(iname), tokdata, "r");
     if (!fp1)
         return CKR_OK;          // no token objects
 
     while (fgets(tmp, 50, fp1)) {
```

Comment:
check for sizeof(tmp) instead of "50"

Reply:
Yes, we had this discussion also when reviewing the token migration tool. There we changed the code similar to:
```
char tmp[PATH_MAX];
while (fgets(tmp, PATH_MAX, fp)) { ... }
```
However, we are just reading token object names here. They only have 8 chars.

Reply:
I guess Juergen did not change this since it is not related to his fix.
Although I agree that it could be enhanced, but please in a different commit.

Reply:
fgets is still on my todo list.  Will change it in a separate commit later.